### ISSUE TITLE
PermanentFail in wget virus

### ISSUE THREAD

**Issue by Samantao93** [yyoshiaki/VIRTUS2#42]
When I execute this in a Conda environment with all the necessary installations, the human FASTA file downloads successfully. However, when it tries to download the virus FASTA file, it results in a "PermanentFail" error. We are using a proxy, which might be causing the issue. Could you provide guidance on how to resolve this?

createindex.cwl https://raw.githubusercontent.com/yyoshiaki/VIRTUS2/master/workflow/createindex.job.yaml

INFO path/.conda/envs/virtus2/bin/cwltool 3.1.20240508115724
INFO Resolved 'path/VIRTUS2/bin/createindex.cwl' to 'file:///path/VIRTUS2/bin/createindex.cwl'
INFO [workflow ] start
INFO [workflow ] starting step wget_virus
INFO [step wget_virus] start
INFO [job wget_virus] /tmp/h0_x3dkw$ docker \
    run \
    -i \
    --mount=type=bind,source=/tmp/h0_x3dkw,target=/TkuKdQ \
    --mount=type=bind,source=/tmp/r0wueiha,target=/tmp \
    --workdir=/TkuKdQ \
    --read-only=true \
    --user=1012:1012 \
    --rm \
    --cidfile=/tmp/bes5i9q6/20240812134745-514374.cid \
    --env=TMPDIR=/tmp \
    --env=HOME=/TkuKdQ \
    inutano/wget:1.20.3-r1 \
    wget \
    --output-document=viruses.fasta \
    https://raw.githubusercontent.com/yyoshiaki/VIRTUS2/master/data/viruses.fasta 2> /tmp/h0_x3dkw/wget-stderr.log
**INFO [job wget_virus] Max memory used: 0MiB
WARNING [job wget_virus] exited with status: 4
WARNING [job wget_virus] completed permanentFail
WARNING [step wget_virus] completed permanentFail
INFO [workflow ] completed permanentFail
{
    "downloaded_virus": {
        "location": "file:///path/virtus/results/viruses.fasta",
        "basename": "viruses.fasta",
        "class": "File",
        "checksum": "sha1$da39a3ee5e6b4b0d3255bfef95601890afd80709",
        "size": 0,
        "path": "path/virtus/results/viruses.fasta"
    },
    "starIndex_virus": null,
    "starIndex_human": null,
    "downloaded_genomefasta_human": null
}WARNING Final process status is permanentFail**

**Comment by yyoshiaki** [comment]
Could you see `/tmp/h0_x3dkw/wget-stderr.log`?
wget looks like it is not working.

**Comment by Samantao93** [comment]
No, because when it stops doing the wget, it removes the file. I tried, but it happens so fast. Yes, it seems that the issue is with wget, but when I tried using wget with the URL directly, the download completed successfully. I'm not sure why it doesn't work with the createindex.cwl.
